current/en-us/5. templating/4. custom-elements.md

```diff
@@ -378,3 +378,405 @@ There are lots of options that allow you to change how custom elements work. The
+      content.setAttribute("show.bind", `activeTabId=='${i}'`);
+      content.append(...Array.from(tab.childNodes));
+      contentTemplate.content.appendChild(content);
+
+      node.removeChild(tab);
+    }
+
+    // Activate the first tab
+    const bindingEngine: BindingEngine = Container.instance.get(BindingEngine);
+    instruction.attributes = {
+      ...instruction.attributes,
+      "active-tab-id": bindingEngine.createBindingExpression("activeTabId", "'0'")
+    };
+
+    node.append(headerTemplate, contentTemplate);
+    return true;
+  }
+
+  @bindable public activeTabId: string;
+  public showTab(tabId: string) { this.activeTabId = tabId; }
+}
+```
+
+The `processTabs` method used in `@processContent` may look overwhelming, but all it does is mostly pure DOM manipulation. The function takes four arguments, out of which `node` holds the DOM fragment for `<tabs>...</tabs>` used in `app.html`. And then in this function, we transforms the original DOM fragment to something like below.
+
+```HTML 
+<tabs>
+  <template replace-part="header">
+    <button click.delegate="showTab('0')">Top Movies</button>
+    <button click.delegate="showTab('1')">Top TV Shows</button>
+    <button click.delegate="showTab('2')">Recently Added</button>
+  </template>
+  <template replace-part="content">
+    <div show.bind="activeTabId=='0'">...</div>
+    <div show.bind="activeTabId=='1'">...</div>
+    <div show.bind="activeTabId=='2'">...</div>
+  </template>
+</tabs>
+```
+
+You must have noted that we have also used the some Aurelia syntaxes such as `click.delegate`, or `show.bind`. We have even used `instruction` to bind the `@bindable activeTabId` of `Tabs` class, by `createBindingExpression` using `BindingEngine`. This enables us to set `'0'` to `activeTabId` which in turn activates the first tab by default. However, we need to instruct Aurelia to process these instructions further which we do by `return`ing `true` at the end of the function. To keep the example simple, we have removed some non-crucial part of the code. However, you can see the result and source code below and play with it.
+
+[simple tab demo](https://codesandbox.io/s/r5v171o1wo?autoresize=1&hidenavigation=1&module=%2Fsrc%2FTabs.js&view=preview)
+
+You maybe thinking that "all that is okay, but how the heck can I use my awesome custom controls??". You can definitely do that. In our example, we are packing all the child nodes of `tabs` directly to the respective content `div`. Therefore, you are absolutely free to use your awesome custom controls to compose the tab content. Let's see if we can use a custom element for the header? This might be good idea, as you start adding some styles to the header, it might start looking bulky and messy (as you might have seen in the code sandbox). Let's start with creating a custom element for the tab header.
+
+```HTML TabHeader.html
+<template>
+	<button class="btn btn-link" class.bind="isActive?'active':''" click.delegate="tabSelector()">
```

Comment:
A little space typo here

Reply:
Tried to resolve it. Maybe it was a tab/space thing. Copied from code sandbox or something like that.